### PR TITLE
Tweak: Notes was added to the right click context menu [ED-7411]

### PR DESCRIPTION
**Pull request by mark-elementor**

### CODE REVIEW

**Review by matipojo — CHANGES_REQUESTED** [review]
Please move to a Notes module

File: assets/dev/js/editor/elements/views/behaviors/context-menu.js [hunk 1]
```diff
@@ -55,6 +62,8 @@ module.exports = Marionette.Behavior.extend( {
 			} );
 		}
 
+		contextMenuGroups = elementor.hooks.applyFilters( 'elements/widget/onInitContextMenuGroups', contextMenuGroups, this );
```
Comment (StyleShit) [inline assets/dev/js/editor/elements/views/behaviors/context-menu.js:65]: It shouldn't be here, this hook is related to widgets, and it's not a widget.
Plus, it won't work in older Pro versions if you add a new hook

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/behaviors/context-menu.js (hunk 1 above)
Comment (mark-elementor) [reply assets/dev/js/editor/elements/views/behaviors/context-menu.js]: Changed.

File: assets/dev/js/editor/elements/views/behaviors/context-menu.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import AddSectionBase	from 'elementor-views/add-section/base';
+import AddSectionBase from 'elementor-views/add-section/base';
```
Comment (StyleShit) [inline assets/dev/js/editor/elements/views/behaviors/context-menu.js:1]: not related to this PR

File: assets/dev/js/editor/regions/panel/pages/menu/menu.js [hunk 3]
```diff
@@ -67,6 +67,7 @@ PanelMenu.addAdminMenu = () => {
 		],
 	}, { at: 0 } );
 
+	// TODO: Move to the Notes module
```
Comment (StyleShit) [inline assets/dev/js/editor/regions/panel/pages/menu/menu.js:70]: why didn't you move it already?

File: modules/notes/assets/js/editor/module.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	onInit() {
+		this.contextMenuNotesGroup();
+	}
+
+	contextMenuNotesGroup() {
+		const elTypes = [ 'widget', 'section', 'column', 'container' ];
+
+		elTypes.forEach( ( type ) => {
+			elementor.hooks.addFilter( `elements/${ type }/contextMenuGroups`, this.contextMenuAddGroup );
+		} );
```
Comment (StyleShit) [inline modules/notes/assets/js/editor/module.js:11]: Maybe add a filter for all of the elements, so you'll have a single filter?
```suggestion
		elementor.hooks.addFilter( `elements/contextMenuGroups`, this.contextMenuAddGroup );
```

Check with @avivu or @shilo-ey if it fine from their side

File: modules/notes/assets/js/editor/module.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	/**
+	 * Append the 'Notes' context menu group
+	 *
+	 * @since 3.8.0
```
Comment (StyleShit) [inline modules/notes/assets/js/editor/module.js:17]: you don't really need this

File: modules/notes/assets/js/editor/module.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+			afterGroupIndex = groups.length;
+		}
+
+		groups.splice( afterGroupIndex, 0, {
+			name: 'notes',
+			actions: [
+				{
+					name: 'open_notes',
+					title: __( 'Notes', 'elementor' ),
+					shortcut: '<i class="eicon-pro-icon" />',
```
Comment (StyleShit) [inline modules/notes/assets/js/editor/module.js:36]:
```suggestion
					shortcut: '<i class="eicon-pro-icon"></i>',
```

Closing tag is required:
https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-i-element

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/behaviors/context-menu.js (hunk 2 above)
Comment (mark-elementor) [reply assets/dev/js/editor/elements/views/behaviors/context-menu.js:1]: I know, but I saw it, so I already fixed it. Shouldn't I?

File: assets/dev/js/editor/regions/panel/pages/menu/menu.js (hunk 3 above)
Comment (mark-elementor) [reply assets/dev/js/editor/regions/panel/pages/menu/menu.js:70]: Very out of scope of the current task.

File: modules/notes/assets/js/editor/module.js (hunk 5 above)
Comment (mark-elementor) [reply modules/notes/assets/js/editor/module.js:17]: The comment?

File: modules/notes/assets/js/editor/module.js (hunk 6 above)
Comment (mark-elementor) [reply modules/notes/assets/js/editor/module.js:36]: Done

File: modules/notes/module.php [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+			true
+		);
+	}
+
+	/**
+	 * Initialize the Container-Converter module.
+	 *
+	 * @return void
+	 */
+	public function __construct() {
```
[the remark this answers is not in the document]
Comment (mark-elementor) [reply modules/notes/module.php:36]: Good idea, done!

File: modules/notes/assets/js/editor/module.js (hunk 4 above)
Comment (mark-elementor) [reply modules/notes/assets/js/editor/module.js:11]: I thought about it when working on it, but I am not sure how good of an idea that is, especially that there are so many places that already have custom mods to the contextMenu.
Touching it (or even starting a discussion over it) might make it from a tiny task into a big thing.
I do agree it's a good idea, but maybe as a TODO for later?

File: modules/notes/assets/js/editor/module.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+					callback: () => {},
+				},
+			],
+		} );
+
+		return groups;
+	}
+}
+
+new Module();
```
[the remark this answers is not in the document]
Comment (mark-elementor) [reply modules/notes/assets/js/editor/module.js]: Done

File: assets/dev/js/editor/regions/panel/pages/menu/menu.js (hunk 3 above)
Comment (StyleShit) [reply assets/dev/js/editor/regions/panel/pages/menu/menu.js:70]: but you're already here. If it's easy to implement, I think you should add it here

File: modules/notes/assets/js/editor/module.js (hunk 5 above)
Comment (StyleShit) [reply modules/notes/assets/js/editor/module.js:17]: the "since", but it's not critical

File: modules/notes/assets/js/editor/module.js (hunk 4 above)
Comment (StyleShit) [reply modules/notes/assets/js/editor/module.js:11]: You don't need to touch the other places, you're just adding a new API.
Anyway, yeah, maybe we should have a new task for that to create that API and update all of the places that uses the old one.

File: assets/dev/js/editor/elements/views/behaviors/context-menu.js (hunk 2 above)
Comment (mark-elementor) [reply assets/dev/js/editor/elements/views/behaviors/context-menu.js:1]: Ok, removed

File: assets/dev/js/editor/regions/panel/pages/menu/menu.js (hunk 3 above)
Comment (mark-elementor) [reply assets/dev/js/editor/regions/panel/pages/menu/menu.js:70]: Not super complicated, but also will require some development and some testing to do.
I don't want to mix with the current task. Created a new one for it.

File: modules/notes/assets/js/editor/module.js (hunk 4 above)
Comment (mark-elementor) [reply modules/notes/assets/js/editor/module.js:11]: Not super complicated, but also will require some development and some testing to do.
I don't want to mix with the current task. Created a new one for it.

**Review by StyleShit — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**